tests/unittests/utils_tests.py

```
@@ -952,6 +953,116 @@ def test_cpucount_exception(self, cpu_count_mock):
+    ):
+        result = fou.get_cpu_count()
+        self.assertEqual(result, 8)
+
+    # =============================================================
+    # test os.cpu_count fallback (mac/windows/other)
+    # =============================================================
+    @patch("builtins.open", side_effect=FileNotFoundError)
+    @patch("os.sched_getaffinity", side_effect=OSError, create=True)
+    @patch("os.cpu_count", return_value=16)
+    def test_fallback_to_cpu_count(
+        self, mock_cpu, mock_affinity, mock_cgroups_file
```

Ruff (0.14.10)

Unused method argument: `mock_cpu`

(ARG002)